### PR TITLE
Adding flag for MultiProcessor Compilation in Premake

### PR DESCRIPTION
**Pull request by ryanbbernstein**
Only works on Visual Studio but premake flags are ignored on unsupported platforms.

### CODE REVIEW

**Comment by CleverSource** [comment]
Seems nice. Since premake flags are ignored on unsupported platforms it shouldn't affect anything. 

You should consider undoing your changes towards ``.gitmodules`` though, it seems your changes from your other PR (#103) managed to get into this PR's branch even though they're on different branches.

File: .gitmodules [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 	path = Hazel/vendor/spdlog
 	url = https://github.com/gabime/spdlog
+	branch = v1.x
 [submodule "Hazel/vendor/GLFW"]
 	path = Hazel/vendor/GLFW
 	url = https://github.com/TheCherno/glfw
 [submodule "Hazel/vendor/imgui"]
 	path = Hazel/vendor/imgui
 	url = https://github.com/TheCherno/imgui
+	branch = docking
```
Comment (CleverSource) [inline .gitmodules]: This should be undone; this PR is for a premake change, not changing git modules.

File: .gitmodules [hunk 2]
```diff
@@ -1,12 +1,14 @@
 [submodule "Hazel/vendor/spdlog"]
 	path = Hazel/vendor/spdlog
 	url = https://github.com/gabime/spdlog
+	branch = v1.x
```
Comment (CleverSource) [inline .gitmodules]: This should be undone; this PR is for a premake change, not changing git modules.

**Comment by ryanbbernstein** [comment]
Sorry about that. I didn't realize pushing would automatically add to the pull request. To remove do i just revert the commit and push?

**Comment by CleverSource** [comment]
@ryanbbernstein Yes.

**Comment by ryanbbernstein** [comment]
ok, its reverted

**Comment by CleverSource** [comment]
Looks good. 

LGTM

**Comment by LovelySanta** [comment]
Didn't have time yesterday to test this.
LGTM

**Pull request merged**